webui/react/src/components/LogViewer.tsx

```diff
@@ -350,6 +374,18 @@ const LogViewer: React.FC<Props> = forwardRef((
     </Space>
   );
 
+  const levelCss = (defaultCss: string, level?: string): string => {
+    const classes = [ defaultCss ];
+    if (level) classes.push(css[level]);
+    return classes.join(' ');
+  };
+
+  const addClipboardPrefix = (log: Log): string => {
+    const content = `${log.time} [${log.level?.toLocaleUpperCase()}] `;
+    const prefix = `<span class=${css.clipboard}>${content}</span>`;
+    return prefix + ansiToHtml(log.message);
```

Comment:
question: is `ansiToHtml` safe? the attack vector is pretty far-fetched (someone who has access to the cluster generates a log statement that breaks into the DOM) but worth asking.

Reply:
great question. It is as safe as the package `ansi-to-html`. `ansiToHtml` escapes out `<` and `>`, so it prevents any `<script>` injections to run even if someone was able to successfully do so. Package `ansi-to-html` takes in the ANSI byte string and converts it to html, but because it only converts ANSI color codes, it should only be able to generate non-script HTML tags.